spring-beans/src/test/java/org/springframework/beans/BeanUtilsTests.java

```diff
@@ -330,6 +331,49 @@ void isNotSimpleProperty(Class<?> type) {
 	private void assertSignatureEquals(Method desiredMethod, String signature) {
 		assertThat(BeanUtils.resolveSignature(signature, MethodSignatureBean.class)).isEqualTo(desiredMethod);
 	}
+	
+	@Test
+	void testCopiedParametersType() {
```

Comment:
This method does not actually test the new support for generic types.

Instead, it verifies that non-matching types are not copied, which was already the case.

Reply:
yes I considered only to not to copy non-matching types as mentioned in issue.